src/components/views/beacon/RoomCallBanner.tsx

```diff
@@ -0,0 +1,180 @@
+        },
+        [call],
+    );
+
+    const onClick = () => {
+        dispatcher.dispatch<ViewRoomPayload>({
+            action: Action.ViewRoom,
+            room_id: roomId,
+            metricsTrigger: undefined,
+            event_id: callEvent.getId(),
+            scroll_into_view: true,
+            highlighted: true,
+        });
+    };
+
+    let [buttonText, buttonKind, onButtonClick] = [null, null, null, null];
+
+    switch (call.connectionState) {
+        case ConnectionState.Disconnected:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Join"),
+                "primary",
+                connect,
+            ];
+            break;
+        case ConnectionState.Connecting:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Join"),
+                "primary",
+                null,
+            ];
+            break;
+        case ConnectionState.Connected:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Leave"),
+                "danger",
+                disconnect,
+            ];
+            break;
+        case ConnectionState.Disconnecting:
+            [buttonText, buttonKind, onButtonClick] = [
+                _t("Leave"),
+                "danger",
+                null,
+            ];
+            break;
+    }
+    if (!call) return <Fragment />;
+
+    return (
+        <div
+            className="mx_RoomLiveShareWarning mx_RoomCallBanner"
+            onClick={onClick}
+        >
+            <div className="mx_RoomCallBanner_text">
+                <span className="mx_RoomCallBanner_label">Video call</span>
+                <CallDurationFromEvent mxEvent={callEvent} />
+            </div>
+
+            <AccessibleButton
+                className="mx_RoomCallBanner_button"
+                data-test-id="room-live-share-primary-button"
+                onClick={onButtonClick}
+                kind={buttonKind}
+                element="button"
+                disabled={false}
+            >
+                { buttonText }
+            </AccessibleButton>
+        </div>
+    );
+};
+
+interface Props {
+    roomId: Room["roomId"];
+}
+
+const RoomCallBanner: React.FC<Props> = ({ roomId }) => {
+    const call = useCall(roomId);
+
+    // this section is to check if we have a live location share. If so, we dont show the call banner
+    const isMonitoringLiveLocation = useEventEmitterState(
+        OwnBeaconStore.instance,
+        OwnBeaconStoreEvent.MonitoringLivePosition,
+        () => OwnBeaconStore.instance.isMonitoringLiveLocation,
+    );
+
+    const liveBeaconIds = useEventEmitterState(
+        OwnBeaconStore.instance,
+        OwnBeaconStoreEvent.LivenessChange,
+        () => OwnBeaconStore.instance.getLiveBeaconIds(roomId),
+    );
+
+    if (isMonitoringLiveLocation && liveBeaconIds.length) {
+        return null;
+    }
+
+    // Check if the call is already showing. No banner is needed in this case.
```

Comment:
I think we should additionally hide the banner if you're connected to the call, since I don't think the extra 'leave' button is necessary - the picture-in-picture view already serves the purpose of keeping a leave button always available on screen during the call